src/store/global/global.thunks.ts

```diff
@@ -895,16 +957,34 @@ function checkPendingDeposits(): AppThunk {
                   })
                   .catch(() => dispatch(globalActions.checkPendingDepositsSuccess()));
               })
-              .catch(() => ({}));
+              .catch((error: unknown) => {
+                const errorMsg = adapters.parseError(error);
+                dispatch(
+                  openSnackbar({
+                    type: "error",
+                    raw: error,
+                    parsed: errorMsg,
+                  })
+                );
+              });
           })
-          .catch(() => ({}));
+          .catch((error: unknown) => {
+            const errorMsg = adapters.parseError(error);
```

Comment:
It really feels like this catch handler can be abstracted. There's a lot of repeated code

Reply:
Agreed. We should work on this abstraction once we're finished and have the whole picture clear.

Reply:
I've installed the lib [`stacktrace-js`](https://www.stacktracejs.com/) to produce meaningful stack traces for the errors using the source maps and as you suggested, abstracted the repetitive code to a global thunk named `processError`.